tests/integration/test_headers_and_subtotals.py

```diff
@@ -989,3 +1004,1143 @@ def test_col_labels_with_top_hs(self):
+                [
+                    1.8313465940e02,
+                    -7.4554570247e01,
+                    np.nan,
+                    -1.2007539044e02,
+                    -1.8313465940e02,
+                ],
+                [
+                    -1.4327127898e02,
+                    np.nan,
+                    3.8997895966e01,
+                    3.2873649739e01,
+                    1.4327127898e02,
+                ],
+                [
+                    -5.4027797535e01,
+                    np.nan,
+                    2.1541918342e01,
+                    7.7287663457e00,
+                    5.4027797535e01,
+                ],
+                [
+                    8.9319544542e01,
+                    np.nan,
+                    -6.4430616588e00,
+                    -2.3624730231e01,
+                    -8.9319544542e01,
+                ],
+                [
+                    -3.7294644763e01,
+                    np.nan,
+                    9.3504419061e00,
+                    3.1011065784e01,
+                    3.7294644763e01,
+                ],
+                [
+                    4.0557613193e02,
+                    np.nan,
+                    -7.4604498097e01,
+                    -4.3802723077e01,
+                    -4.0557613193e02,
+                ],
+                [np.nan, np.nan, 7.5419970636e-01, -2.2727871868e01, np.nan],
+                [np.nan, np.nan, np.nan, -7.1657702639e01, np.nan],
+                [
+                    6.2248620699e02,
+                    -1.8416144299e02,
+                    np.nan,
+                    -1.7558752705e02,
+                    -6.2248620699e02,
+                ],
+                [np.nan, -1.0646207700e01, np.nan, np.nan, np.nan],
+                [np.nan, -3.9963319461e01, np.nan, np.nan, np.nan],
+                [np.nan, -3.2226859413e01, np.nan, np.nan, np.nan],
+                [np.nan, -2.0011396180e01, np.nan, np.nan, np.nan],
+                [np.nan, -1.8789849856e01, np.nan, np.nan, np.nan],
+                [np.nan, -5.2178782695e01, np.nan, np.nan, np.nan],
+                [
+                    -8.6951552411e01,
+                    2.6820253146e00,
+                    1.1988726047e01,
+                    np.nan,
+                    8.6951552411e01,
+                ],
+                [
+                    -9.0848015208e01,
+                    4.5442739110e00,
+                    1.4191957607e01,
+                    np.nan,
+                    9.0848015208e01,
+                ],
+                [
+                    -2.3435847246e01,
+                    -3.6990146410e-01,
+                    np.nan,
+                    1.1825276005e01,
+                    2.3435847246e01,
+                ],
+            ],
+        )
+
+        # Test pvals
+        np.testing.assert_almost_equal(
+            slice_.pvals,
+            [
+                [0.00000000e00, np.nan, 0.00000000e00, np.nan, 0.00000000e00],
```

Comment:
This seems unnecessary :) Can you try and make it with just normal zeros? Maybe even use the reshape, so you only focus on important entries, like so:
```
np.array([0, np.nan, 0, np.nan, 0] * 10).reshape((10, 5))
```
or something similar...